search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java

```diff
@@ -105,20 +106,25 @@ public ElasticSearchHighLevelRestClient(Environment environment, RestHighLevelCl
      */
     public SearchResourcesResponse searchSingleTerm(String term, int results) throws ApiGatewayException {
         try {
-            QueryStringQueryBuilder queryBuilder = QueryBuilders.queryStringQuery(term);
-            final SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
-                .query(queryBuilder)
-                .size(results);
-            final SearchRequest searchRequest = new SearchRequest(elasticSearchEndpointIndex);
-            searchRequest.source(sourceBuilder);
-            logger.debug(SEARCH_REQUEST_MESSAGE, elasticSearchEndpointIndex, term, searchRequest);
-            SearchResponse searchResponse = elasticSearchClient.search(searchRequest, RequestOptions.DEFAULT);
+            SearchResponse searchResponse = doSearch(term, results);
             return toSearchResourcesResponse(searchResponse.toString());
         } catch (Exception e) {
             throw new SearchException(e.getMessage(), e);
         }
     }
 
+    private SearchResponse doSearch(String term, int results) throws IOException {
+        QueryStringQueryBuilder queryBuilder = QueryBuilders.queryStringQuery(term);
+        final SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+            .query(queryBuilder)
+            .size(results);
+        final SearchRequest searchRequest = new SearchRequest(elasticSearchEndpointIndex);
+        searchRequest.source(sourceBuilder);
+        logger.debug(SEARCH_REQUEST_MESSAGE, elasticSearchEndpointIndex, term, searchRequest);
+        SearchResponse searchResponse = elasticSearchClient.search(searchRequest, RequestOptions.DEFAULT);
+        return searchResponse;
+    }
+
```

Comment:
Possibly (although naming is weak)

```

   private SearchResponse doSearch(String term, int results) throws IOException {
        final SearchRequest searchRequest = new SearchRequest(elasticSearchEndpointIndex);
        searchRequest.source(getSearchSourceBuilder(term, results));
        logSearch(term, searchRequest);
        return elasticSearchClient.search(searchRequest, RequestOptions.DEFAULT);
    }

    private void logSearch(String term, SearchRequest searchRequest) {
        logger.debug(SEARCH_REQUEST_MESSAGE, elasticSearchEndpointIndex, term, searchRequest);
    }

    private SearchSourceBuilder getSearchSourceBuilder(String term, int results) {
        return new SearchSourceBuilder()
                .query(queryStringQuery(term))
                .size(results);
    }
```

Reply:
extraced searchRequest ganeration, removed logging without interest anymore